### DISCUSSION TITLE
nvidia-snatcher as a service using node-windows

### DISCUSSION THREAD

**Discussion by derekhansell** [coreybutler/node-windows#287] [Q&A] [answered]
Hey there!

Total node.js n00b here. I'm trying to get nvidia-snatcher installed as a service on a Windows Server. I used the readme example script as a base, and it installed a service. But the service won't stay running.

Script to install service:

```
var Service = require('node-windows').Service;

// Create a new service object
var svc = new Service({
  name:'nvidia-snatcher',
  description: 'Scrapes websites for Nvidia GPU stock.',
  script: 'C:\git\nvidia-snatcher\src\index.ts',
//  nodeOptions: [
//    '--harmony',
//    '--max_old_space_size=4096'
//  ]
  //, workingDirectory: '...'
  //, allowServiceLogon: true
});

// Listen for the "install" event, which indicates the
// process is available as a service.
svc.on('install',function(){
  svc.start();
});

svc.install();
```

error log:

> internal/modules/cjs/loader.js:888
>   throw err;
>   ^
> 
> Error: Cannot find module 'C:\git\nvidia-snatcher\git
> 
> vidia-snatchersrcindex.ts'
>     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:885:15)
>     at Function.Module._load (internal/modules/cjs/loader.js:730:27)
>     at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
>     at internal/main/run_main_module.js:17:47 {
>   code: 'MODULE_NOT_FOUND',
>   requireStack: []
> }

Kinda lost...

**Comment by coreybutler** [comment, marked as answer]
The major issue here is Node.js doesn't support TypeScript, so you cannot reference a `.ts` file directly.
